rapid7_insightvm/komand_rapid7_insightvm/util/resource_helper.py

```diff
@@ -120,54 +153,63 @@ def resource_request(self, endpoint: str, method: str = 'get', params: dict = No
 
         return resource
 
-    def paged_resource_request(self, endpoint: str, method: str = 'get', params: dict = None,
-                               payload: dict = None) -> list:
+    def paged_resource_request(self, endpoint: str, method: str = 'get', params: Collection = None,
+                               payload: dict = None, number_of_results: int = 0) -> list:
         """
         Fetches all resources from a paged APIv3 endpoint
         :param endpoint: Endpoint for the APIv3 call defined in endpoints.py
         :param method: HTTP method for the API request
         :param params: URL parameters to append to the request
         :param payload: JSON body for the API request if required
+        :param number_of_results: The total number of results to retrieve. 0 is unlimited
         :return: List object of API resources
         """
         resources = []
         current_page = 0
+        results_retrieved = 0
+        last_page = False
 
         # Handle various scenarios where params may be passed
         if not params:
             params = {
-                "size": 500,
-                "page": current_page
+                'size': 500,
+                'page': current_page
             }
+        if params == list:
+            parameters = RequestParams.from_tuples(params)
         else:
-            if 'size' not in params:
-                params['size'] = 500
-            # Enable requests with arbitrary starting points
-            if 'page' in params:
-                current_page = params['page']
+            parameters = RequestParams.from_dict(params)
 
         while True:
-            self.logger.info(f"Fetching results from page {current_page}")
-            params['page'] = current_page
-            response = self._get_resource_page(endpoint=endpoint,
-                                               method=method,
-                                               params=params,
-                                               payload=payload)
+            self.logger.info(f'Fetching results from page {current_page}')
+            parameters['page'] = [current_page]
```

Comment:
```suggestion
            parameters['page'] = current_page
```
Why are you putting the current page into a list?